apps/api/src/app/workflows-v2/usecases/upsert-workflow/upsert-workflow.usecase.ts

```diff
@@ -123,15 +123,13 @@ export class UpsertWorkflowUseCase {
     command: UpsertWorkflowCommand,
     workflow: NotificationTemplateEntity
   ): Promise<GetPreferencesResponseDto | undefined> {
-    if (!command.workflowDto.preferences?.user) {
-      return undefined;
-    }
-    await this.upsertPreferences(workflow, command);
+    await this.upsertUserWorkflowPreferences(workflow, command);
+    await this.upsertWorkflowPreferences(workflow, command);
```

Comment:
Added this to always insert both user and workflow preferences, in case they're not provided, we create default ones.

Reply:
thought: agree on `user` preference default upsert but, i wonder if we need to create a default `workflow` preference for users that do not use code-first, feels a bit redundant and misleading data.